galley/pkg/config/analysis/analyzers/serviceentry/httpsonhttp.go

```diff
@@ -0,0 +1,119 @@
+	return analysis.Metadata{
+		Name:        "serviceentry.HTTPSOnHTTPAnalyzer",
+		Description: "Checks if HTTPS traffic on HTTP port",
+		Inputs: collection.Names{
+			collections.IstioNetworkingV1Alpha3Serviceentries.Name(),
+			collections.IstioNetworkingV1Alpha3Destinationrules.Name(),
+		},
+	}
+}
+
+func (serviceEntry *HTTPSOnHTTPAnalyzer) Analyze(context analysis.Context) {
+	context.ForEach(collections.IstioNetworkingV1Alpha3Serviceentries.Name(), func(resource *resource.Instance) bool {
+		serviceEntry.analyzeProtocol(resource, context)
+		return true
+	})
+}
+
+func (serviceEntry *HTTPSOnHTTPAnalyzer) analyzeProtocol(resource *resource.Instance, context analysis.Context) {
+	se := resource.Message.(*v1alpha3.ServiceEntry)
+
+	for _, host := range se.Hosts {
+		relatedDestinationRules := serviceEntry.getDestinationRules(host, context)
+
+		for index, port := range se.Ports {
+			if port.Protocol == "HTTP" {
+				usedPort := port.Number
+				if port.TargetPort != 0 {
+					usedPort = port.TargetPort
+				}
+
+				if usedPort == 443 {
+					if len(relatedDestinationRules) == 0 {
+						serviceEntry.sendMessage(resource, host, index, context)
+					}
+
+					if !serviceEntry.checkDestinationRulesPerformTLSOrigination(relatedDestinationRules, usedPort, context) {
+						serviceEntry.sendMessage(resource, host, index, context)
+					}
+				}
+			}
+		}
+	}
+}
+
+func (serviceEntry *HTTPSOnHTTPAnalyzer) getDestinationRules(host string, context analysis.Context) []resource.FullName {
+	var destinationRules []resource.FullName
+
+	context.ForEach(collections.IstioNetworkingV1Alpha3Destinationrules.Name(), func(r *resource.Instance) bool {
+		destinationRule := r.Message.(*v1alpha3.DestinationRule)
+
+		if host == destinationRule.Host {
+			destinationRules = append(destinationRules, r.Metadata.FullName)
+		}
+
+		return true
+	})
+
+	return destinationRules
+}
+
+func (serviceEntry *HTTPSOnHTTPAnalyzer) checkDestinationRulesPerformTLSOrigination(
+	destinationRulesName []resource.FullName,
+	port uint32,
+	context analysis.Context) bool {
+	for _, destinationRuleName := range destinationRulesName {
+		destinationRuleInstance := context.Find(collections.IstioNetworkingV1Alpha3Destinationrules.Name(), destinationRuleName)
+		destinationRule := destinationRuleInstance.Message.(*v1alpha3.DestinationRule)
+		for _, portLevelSetting := range destinationRule.TrafficPolicy.PortLevelSettings {
+			if portLevelSetting.Port.Number == port && portLevelSetting.Tls.Mode != 0 {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func (serviceEntry *HTTPSOnHTTPAnalyzer) sendMessage(resource *resource.Instance, host string, index int, context analysis.Context) {
+	message := msg.NewServiceEntryHTTPSTrafficOnHTTPPort(resource, host)
+
+	if line, ok := util.ErrorLine(resource, fmt.Sprintf(util.ServiceEntryPort, index)); ok {
+		message.Line = line
+	}
+
+	context.Report(collections.IstioNetworkingV1Alpha3Serviceentries.Name(), message)
```

Comment:
Does it make sense to report this only on the ServiceEntry, or should it also be reported on the DestinationRule with a similar, perhaps slightly different, message?

Reply:
I am not sure, I believe a single analyzer is already good enough to inform this right? adding the same analyzer for DestinationRule will increase calculation time when executing the analyzer. wdyt?

Reply:
If the user has turned on Status messages when installing the control plane, it makes sense to set the status on both the SE and DR so that tools can display it.  Regarding performance: if you do this, don't write logic to check DRs.  Instead have `checkDestinationRulesPerformTLSOrigination()` return the conflicting DR, and then just call `context.Report()` it in addition to calling it for the SE.